mu4/notation/internal/notationselection.h

```diff
@@ -38,17 +38,36 @@ class NotationSelection : public INotationSelection
     bool isNone() const override;
     bool isRange() const override;
 
+    bool canCopy() const override;
+    QMimeData* mimeData() const override;
+
     Element* element() const override;
     std::vector<Element*> elements() const override;
 
     SelectionRange range() const override;
 
     QRectF canvasBoundingRect() const override;
 
-private:
+    std::vector<QRectF> rangeBoundingArea() const override;
 
+private:
     Ms::Score* score() const;
 
+    Ms::Segment* rangeStartSegment() const;
+    Ms::Segment* rangeEndSegment() const;
+
+    int selectionLastVisibleStaff() const;
+
+    struct RangeSection {
+        const Ms::System* system = nullptr;
+        const Ms::Segment* startSegment = nullptr;
+        const Ms::Segment* endSegment = nullptr;
+    };
+    std::vector<RangeSection> splitRangeBySections(const Ms::Segment* rangeStartSegment, const Ms::Segment* rangeEndSegment) const;
```

Comment:
Yeah, there is a lot of "range" staff here and since all methods are private it makes sense to keep it in separate class

Reply:
thanks!
fixed